tests/handlers/test_oidc.py

```diff
@@ -68,31 +71,42 @@
 }
 
 
+T = TypeVar("T")
+
+
 class TestMappingProvider:
     @staticmethod
-    def parse_config(config):
-        return
+    def parse_config(config: JsonDict) -> None:
+        return None
 
-    def __init__(self, config):
+    def __init__(self, config: T):
         pass
```

Comment:
What does `T` represent here? I don't think we use it elsewhere?

(Maybe MappingProviders are expected to be generic somehow??)

Reply:
Ahh, you noticed this in https://github.com/matrix-org/synapse/pull/14680#discussion_r1049687548

Reply:
So `parse_config` is given the YAML config and it returns something that `__init__` has to take. So the return value of `parse_config` needs to match the `config` variable of `__init__`. I think just hard-coding the `None` type for tests is OK?

Reply:
SGTM!